### PR TITLE
↔️ Fix CB in GRPO

### PR DESCRIPTION
**Pull request by qgallouedec**
Padding side was wrong, this PR is a fix
A few cleanup, late review for #3471

cc @ArthurZucker for viz

### CODE REVIEW

File: trl/trainer/grpo_trainer.py [hunk 1]
```diff
@@ -1217,29 +1218,31 @@ def _generate_and_score_completions(
...
                     )
             completion_ids = [output.generated_tokens for output in all_outputs.values()]
             completion_ids = [torch.tensor(ids, device=device) for ids in completion_ids]
-            completion_ids = pad(completion_ids, padding_value=self.processing_class.pad_token_id)
+            completion_ids = pad(
+                completion_ids, padding_value=self.processing_class.pad_token_id, padding_side="right"
+            )
             prompt_ids = [torch.tensor(ids, device=device) for ids in prompt_inputs.input_ids]
-            prompt_ids = pad(prompt_ids, padding_value=self.processing_class.pad_token_id)
+            prompt_ids = pad(prompt_ids, padding_value=self.processing_class.pad_token_id, padding_side="left")
```
Comment (qgallouedec) [inline trl/trainer/grpo_trainer.py:1229]: Padding side must be `"left"` for prompts, otherwise you'd get (incorrect):

```
[[P01, P01, P03, PAD, PAD, C01, C02, C03, C04],
 [P11, P11, P13, P14, P15, C11, C12, PAD, PAD]]
```

instead of (correct)

```
[[PAD, PAD, P01, P01, P03, C01, C02, C03, C04],
 [P11, P11, P13, P14, P15, C11, C12, PAD, PAD]]
```

File: trl/trainer/grpo_trainer.py [hunk 2]
```diff
@@ -1217,29 +1218,31 @@ def _generate_and_score_completions(
...
-                        prompt_ids, attention_mask=prompt_mask, generation_config=self.generation_config
-                    )
+            with (
+                profiling_context(self, "transformers.generate"),
+                unwrap_model_for_generation(
+                    self.model_wrapped, self.accelerator, gather_deepspeed3_params=self.args.ds3_gather_for_generation
+                ) as unwrapped_model,
+                torch.no_grad(),
+                FSDP.summon_full_params(self.model_wrapped, recurse=False) if self.is_fsdp_enabled else nullcontext(),
+            ):
```
Comment (qgallouedec) [inline trl/trainer/grpo_trainer.py:1242]: simplification, using 

```python
with c1, c2:
```

instead of

```python
with c1:
    with c2:
```

**Comment by kashif** [comment]
good catch! lets also pin the transformer to the latest patch for the generation length fix?

**Comment by kashif** [comment]
can you kindly test with `transformers>=4.53.2`

**Comment by qgallouedec** [comment]
> can you kindly test with transformers>=4.53.2

done in 17157c4

File: trl/trainer/grpo_trainer.py (hunk 1 above)
Comment (shirinyamani) [reply trl/trainer/grpo_trainer.py:1229]: then how we got reasonable output till now? 🧐

Comment (qgallouedec) [reply trl/trainer/grpo_trainer.py:1229]: the error is only when using CB, introduced in #3471
